simtools/data_model/metadata_model.py

```diff
@@ -1,281 +1,178 @@
+    Raises
+    ------
+    FileNotFoundError
+        if schema file is not found
 
     """
-    return {
-        "SITE": {"type": "str", "required": True, "default": None},
-        "CLASS": {"type": "str", "required": True, "default": None},
-        "TYPE": {"type": "str", "required": True, "default": None},
-        "SUBTYPE": {"type": "str", "required": False, "default": None},
-        "ID": {"type": "str", "required": True, "default": None},
-    }
 
+    if schema_file is None:
+        with files("simtools").joinpath("schemas/metadata.schema.yml").open(
+            "r", encoding="utf-8"
+        ) as file:
+            schema = yaml.safe_load(file)
+        _logger.info("Loading schema from simtools package.")
+    else:
+        schema = gen.collect_data_from_yaml_or_dict(in_yaml=schema_file, in_dict=None)
+        _logger.info(f"Loading schema from {schema_file}")
 
-def _category_activity():
-    """
-    Metadata field ACTIVITY.
+    return schema
 
-    Describes the software used to process these data.
 
-    """
-    return {
-        "NAME": {"type": "str", "required": True, "default": None},
-        "TYPE": {"type": "str", "required": True, "default": "software"},
-        "ID": {"type": "str", "required": True, "default": None},
-        "START": {"type": "datetime", "required": False, "default": None},
-        "SOFTWARE": {
-            "NAME": {"type": "str", "required": True, "default": None},
-            "VERSION": {"type": "str", "required": True, "default": None},
-        },
-    }
-
-
-def top_level_reference_schema():
+def top_level_reference_schema(schema_file=None):
     """
     Reference schema following the CTA Top-Level Data Model.
 
     This metadata schema is used for simtools data products.
 
+    Parameters
+    ----------
+    schema_file: str
+        Schema file (jsonschema format) used for validation
+
     Returns
     -------
     dict with reference schema
 
 
     """
 
-    _ref_schema = {
-        "CTA": {
-            "REFERENCE": _category_reference(),
-            "CONTACT": _category_contact(),
-            "PRODUCT": _category_product(),
-            "INSTRUMENT": _category_instrument(),
-            "PROCESS": _category_process(),
-            "ACTIVITY": _category_activity(),
-            "CONTEXT": {
-                "SIM": {
-                    "ASSOCIATION": [
-                        _category_instrument(),
-                    ],
-                    "DOCUMENT": [
-                        _category_document(),
-                    ],
-                },
-            },
-        }
-    }
-    _ref_schema = _metadata_dict_with_defaults(_ref_schema)
-    return _remove_empty_lists(_ref_schema)
-
-
-def metadata_input_reference_schema():
-    """
-    Reference data model scheme for input metadata.
-    Describes metadata provided for input to simtools applications.
+    schema = load_schema(schema_file)
+    return fill_defaults(schema["definitions"]["CTA"])
```

Comment:
I understand. If it is your choice I am fine with that. Maybe in the future, in case the the module gets more complicated we think about changing. Thanks for the explanation!